scripts/validate_bare_pytest_raise.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+"""
+GH #23922
+
+Check for the use of bare pytest raise.
+
+For example:
+
+>>> with pytest.raise(ValueError):
+...    # Some code that raises ValueError
+
+Instead of:
+
+>>> with pytest.raise(ValueError, match="foo"):
+...    # Some code that raises ValueError
+"""
+
+import argparse
+import os
+import sys
+import token
+import tokenize
+from typing import Generator, List, Tuple
+
+FILE_EXTENSIONS_TO_CHECK = ".py"
+
+
+def main(source_path: str, output_format: str) -> bool:
+    """
+    Main entry point of the script.
+
+    Parameters
+    ----------
+    source_path : str
+        Source path representing path to a file/directory.
+    output_format : str
+        Output format of the script.
+
+    Returns
+    -------
+    bool
+        True if found any bare pytest raises.
+
+    Raises
+    ------
+    ValueError
+        If the `source_path` is not pointing to existing file/directory.
+    """
+    if not os.path.exists(source_path):
+        raise ValueError(
+            "Please enter a valid path, pointing to a valid file/directory."
+        )
+
+    is_failed: bool = False
+
+    msg = "Bare pytests raise have been found."
+
+    if os.path.isfile(source_path):
+        for source_path, line_number in bare_pytest_raise(source_path):
+            is_failed = True
+            print(
+                output_format.format(
+                    source_path=source_path, line_number=line_number, msg=msg
+                )
+            )
+
+    for subdir, _, files in os.walk(source_path):
```

Comment:
i think it would be nice to encapsulate most of the boiler plate in a common file used in the other validate_* scripts. can you do this as a pre-cursor PR

Reply:
That is exactly what I was a about to ask, if I should do so.

The only question I have is how the script should be activated?
for example like this:
```
./scripts/validate_patterns.py --bare-pytest --concat-strings pandas
```

or 
```
./scripts/validate_patterns.py --test-cases="barepytest,concatstrins" pandas
```